create-aleo-app/template-extension/static/manifest.json

```diff
@@ -0,0 +1,10 @@
+{
+  "manifest_version": 2,
```

Comment:
Manifest Version 2 is no longer supported for new chromium-based extensions. V2 only works for Firefox extensions.

In V3, web workers can't be spawned from an extension's background script. The API isn't available.

https://bugs.chromium.org/p/chromium/issues/detail?id=1219164&q=extension%20web%20workers&can=2

Reply:
@jonandgon Yup, like @onetrickwolf said, there's a lot of issues and incompatibilities with V3 extensions. V3 extensions had completely broken Wasm, among other things. There's not much point in shipping an example which doesn't even work.

Reply:
Also, V2 extensions are still supported in Firefox, so there's still value in providing a V2 example.

And Firefox will be making various changes to the V3 format in order to properly support various use-cases:

https://blog.mozilla.org/addons/2022/05/18/manifest-v3-in-firefox-recap-next-steps/

Unfortunately the Chrome team is far less willing to make any changes, and so many extensions will be locked into Firefox only.

Reply:
It's possible to use wasm inside a V3 extension. We are using a modified version of the Aleo wasm inside our extension wallet.

Multithreading, however, is not possible. The only workaround that _may_ enable a second thread is the Offscreen Documents API. Take a look:

https://developer.chrome.com/blog/Offscreen-Documents-in-Manifest-v3/

Reply:
And yeah I agree, but chromium-based browsers are the most popular by far. They'll need support if you want widespread user adoption.

Reply:
Of course we'll do our best to support as many environments as possible (including Chrome extensions). I myself am a long-time extension developer, and I've made many Firefox / Chrome extensions.

In fact I started making Chrome extensions all the way back in 2009, when Chrome first implemented extensions. So I'm one of the oldest Chrome extension developers.

And I'm also a co-founder of Rust Wasm, and I was the first person in the world to use Rust Wasm to create Chrome / Firefox extensions. So I'm very familiar with using hacky workarounds to fix various Chrome issues.

But if the Chrome team breaks things and refuses to fix it, then there's nothing we can do about it. I've spent many years arguing with the Chrome team about various stupid decisions that they make, but they very rarely listen to feedback from extension developers. They have the attitude that "they know better", they think that they are smarter than extension developers.

Unfortunately, with V3 it seems that Chrome is intentionally trying to destroy the extension ecosystem, because Google is trying to get rid of ad-blockers. And so in their insane attempt to destroy ad-blockers, they are destroying the entire extension ecosystem as collateral damage.

The Chrome extension ecosystem is really bad, Firefox extensions are so much better. But as you said, Chrome is the most popular so you have to support it despite all the major problems.

Reply:
It's very unfortunate the route Chrome has taken, but I am so glad to see that Aleo is working with OGs in the extension space.